include/deal.II/base/index_set.h

```diff
@@ -79,6 +89,188 @@ class IndexSet
   typedef signed int value_type;
 
   /**
+   * Dereferencing an IntervalIterator will return a reference to an object of this type.
+   */
+  class IntervalAccessor
```

Comment:
I think you want a sentence in the documentation that states what the class actually is, not just what it returns when dereferenced.

None of the members currently have any documentation.